indy_node/server/client_authn.py

```diff
@@ -1,35 +1,41 @@
 from copy import deepcopy
 from hashlib import sha256
 
-from plenum.common.exceptions import UnknownIdentifier
 from plenum.common.types import OPERATION
-from plenum.common.constants import TXN_TYPE, RAW, ENC, HASH, VERKEY
-from plenum.server.client_authn import NaclAuthNr
+from plenum.common.constants import TXN_TYPE, RAW, ENC, HASH
+from plenum.server.client_authn import NaclAuthNr, CoreAuthNr, CoreAuthMixin
 
-from indy_common.constants import ATTRIB
+from indy_common.constants import ATTRIB, POOL_UPGRADE, SCHEMA, CLAIM_DEF, \
+    GET_NYM, GET_ATTR, GET_SCHEMA, GET_CLAIM_DEF, POOL_CONFIG
 from indy_node.persistence.idr_cache import IdrCache
-# from indy_node.persistence.state_tree_store import StateTreeStore
 
 
-class TxnBasedAuthNr(NaclAuthNr):
+class LedgerBasedAuthNr(CoreAuthMixin, NaclAuthNr):
     """
     Transaction-based client authenticator.
     """
 
+    write_types = CoreAuthMixin.write_types.union({ATTRIB, SCHEMA, CLAIM_DEF,
+                                                   POOL_CONFIG, POOL_UPGRADE})
+    query_types = CoreAuthMixin.query_types.union({GET_NYM, GET_ATTR, GET_SCHEMA,
```

Comment:
I think we already have a list with query txns: `openTxns` in `constants/py`. Let's use just one list in all places.

Reply:
`openTxns` is meant for txns like `GET_TXNS`, i have made that change

Reply:
Why can't we move openTxns here as well? so, we have all logic related to signature verification skipping in one place?